docs/CONTRIBUTING.md

```diff
@@ -95,60 +95,125 @@ You have to commit the changes to go.mod, go.sum and before submitting the pull
+- lvm2 packet installed on host machine
+- kubectl/helm
+
+#### Build
+
+##### Set Variables
+
+```
+export REGISTRY=<your_docker_hub>
+export CSI_BAREMETAL_DIR=<full_path_to_csi_baremetal_src>
+export CSI_BAREMETAL_OPERATOR_DIR=<full_path_to_csi_baremetal_operator_src>
+```
+
+##### Build csi-baremetal
+
 ```
+cd ${CSI_BAREMETAL_DIR}
+export CSI_VERSION=`make version`
+
+# Get dependencies
+make dependency
+
+# Compile proto files
 make install-compile-proto
+
+# Generate CRD
 make install-controller-gen
 make generate-deepcopy
-make dependency
-export DRIVE_MANAGER_TYPE=loopbackmgr
-```
-Build binaries:
-```
+
+# Run unit tests
+make test
+
+# Run sanity tests
+make test-sanity
+
+# Run linting
+make lint
+
+# Clean previous artefacts
+make clean
+
+# Build binary
 make build
+make DRIVE_MANAGER_TYPE=loopbackmgr build-drivemgr
+
+# Build docker images
+make download-grpc-health-probe
+make images REGISTRY=${REGISTRY}
+make DRIVE_MANAGER_TYPE=loopbackmgr image-drivemgr REGISTRY=${REGISTRY}
 ```
-Build docker images:
-```
-export REGISTRY="myregistry.com"
-make images
-```
-Run Unit tests:
+
+##### Build csi-baremetal-operator
+
 ```
+cd ${CSI_BAREMETAL_OPERATOR_DIR}
+export CSI_OPERATOR_VERSION=`make version`
+
+# Run unit tests
 make test
-```
-| Action                | Command       | Comment                                                              |
-|-----------------------|---------------|----------------------------------------------------------------------|
-| clean build artifacts | `make clean`  | [`build/_output/baremetal_csi`](./build/_output/baremetal_csi/) directory with all artifacts will be removed |
-| build plugin binary   | `make build`  | artifacts can be found in the [`build/_output/baremetal_csi`](./build/_output/baremetal_csi/) directory.     |
-| build plugin image    | `make images`  | |
-| run linter            | `make lint`  | results will be printed to your terminal|
 
+# Run linting
+make lint
 
+# Build docker image
+make docker-build REGISTRY=${REGISTRY}
+```
 
-##### Running Baremetal CSI E2E tests locally
+##### Prepare kind cluster
 
-* Install `lvm2` package on your machine
-* Create kind (version >= v0.7.0) cluster with the specified config. *Note that kind workers must be run with host IPC*
-```
-kind create cluster --config  test/kind/kind.yaml
 ```
-* KIND can't pull images from remote repository, to load images to local docker repository on nodes:
-```
-export CSI_VERSION=`make version`
-export REGISTRY="asdrepo.isus.emc.com:9042"
+cd ${CSI_BAREMETAL_DIR}
+
+# Build custom kind binary
+make kind-build
 
-make kind-pull-images TAG=${CSI_VERSION} REGISTRY=${REGISTRY}
+# Create kind cluster
+make kind-create-cluster
+
+# Check cluster
+kubectl cluster-info --context kind-kind
+
+# If you use another path for kubeconfig or don't set KUBECONFIG env
+kind get kubeconfig > <path_to_kubeconfig>
+
+# Prepare sidecars 
+# If you have sidecar images pushed into your registary
+make kind-pull-sidecar-images
```

Comment:
Those steps are obsolete.

This is actual code:
```
# Loading sidecars included
make kind-pull-images TAG=${CSI_VERSION} REGISTRY=${REGISTRY}
make kind-tag-images TAG=${CSI_VERSION} REGISTRY=${REGISTRY}
make kind-load-images TAG=${CSI_VERSION} REGISTRY=${REGISTRY}
```

Reply:
`make kind-pull-images` pulls all images from Registry. We have build them locally and want to check. Why we need this?